src/sources/kubernetes_logs/mod.rs

```diff
@@ -239,6 +239,13 @@ pub struct Config {
     #[configurable(derived)]
     #[serde(default)]
     internal_metrics: FileInternalMetricsConfig,
+
+    /// How long to keep an open handle to a rotated log file.
+    /// The default value represents "no limit"
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    #[configurable(metadata(docs::type_unit = "seconds"))]
+    #[serde(default = "default_rotate_wait_sec")]
+    rotate_wait_sec: Duration,
```

Comment:
Same suggestion as above.

```suggestion
    /// How long to keep an open handle to a rotated log file.
    /// The default value represents "no limit".
    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
    #[configurable(metadata(docs::type_unit = "seconds"))]
    #[serde(default = "default_rotate_wait", rename = "rotate_wait_secs")]
    pub rotate_wait: Duration,
```